internal/venice-common/src/main/java/com/linkedin/venice/helix/StringAnnotatedHelixReadOnlySchemaRepository.java

```diff
@@ -0,0 +1,150 @@
+
+  @Override
+  public Collection<SchemaEntry> getValueSchemas(String storeName) {
+    return internalSchemaRepo.getValueSchemas(storeName);
+  }
+
+  @Override
+  /**
+   * Retrieve the superset schema (if exists) or the latest value schema of a store and annotate its map fields.
+   * The annotation will be done everytime when this method is invoked, as the superset schema or the latest value schema
+   * of the store can be changing from time to time.
+   */
+  public SchemaEntry getSupersetOrLatestValueSchema(String storeName) {
+    SchemaEntry schemaEntry = internalSchemaRepo.getSupersetOrLatestValueSchema(storeName);
+    return getAnnotatedStringMapValueSchemaEntry(schemaEntry);
+  }
+
+  @Override
+  /**
+   * Retrieve the superset schema (if exists) and annotate its map fields.
+   * The annotation will be done everytime when this method is invoked, as the superset schema of the store can be
+   * changing from time to time.
+   */
+  public Optional<SchemaEntry> getSupersetSchema(String storeName) {
+    Optional<SchemaEntry> schemaEntryOptional = internalSchemaRepo.getSupersetSchema(storeName);
+    if (schemaEntryOptional.isPresent()) {
+      SchemaEntry schemaEntry = schemaEntryOptional.get();
+      return Optional.of(getAnnotatedStringMapValueSchemaEntry(schemaEntry));
+    } else {
+      return schemaEntryOptional;
+    }
+  }
+
+  @Override
+  public Pair<Integer, Integer> getDerivedSchemaId(String storeName, String derivedSchemaStr) {
+    return internalSchemaRepo.getDerivedSchemaId(storeName, derivedSchemaStr);
+  }
+
+  @Override
+  /**
+   * Retrieve partial update schema of a store and annotate its map fields. The annotation will only be done once in the
+   * repository's lifetime as the result is cached.
+   */
+  public DerivedSchemaEntry getDerivedSchema(String storeName, int valueSchemaId, int partialUpdateSchemaId) {
+    String derivedSchemaId = valueSchemaId + "-" + partialUpdateSchemaId;
+    if (!(partialUpdateSchemaEntryMapCache.containsKey(storeName)
+        && partialUpdateSchemaEntryMapCache.get(storeName).containsKey(derivedSchemaId))) {
+      DerivedSchemaEntry derivedSchemaEntry =
+          internalSchemaRepo.getDerivedSchema(storeName, valueSchemaId, partialUpdateSchemaId);
+      DerivedSchemaEntry annotatedEntry = getAnnotatedStringMapDerivedSchemaEntry(derivedSchemaEntry);
+      partialUpdateSchemaEntryMapCache.putIfAbsent(storeName, new VeniceConcurrentHashMap<>());
+      partialUpdateSchemaEntryMapCache.get(storeName).put(derivedSchemaId, annotatedEntry);
+    }
+    return partialUpdateSchemaEntryMapCache.get(storeName).get(derivedSchemaId);
+  }
+
+  @Override
+  public Collection<DerivedSchemaEntry> getDerivedSchemas(String storeName) {
+    return internalSchemaRepo.getDerivedSchemas(storeName);
+  }
+
+  @Override
+  public DerivedSchemaEntry getLatestDerivedSchema(String storeName, int valueSchemaId) {
+    return internalSchemaRepo.getLatestDerivedSchema(storeName, valueSchemaId);
```

Comment:
Are these methods being used by DCR? The returned schemas are raw schemas, and to avoid mistake, I prefer to throw exception in these methods if they are not used, otherwise, we need to annotate them.

Reply:
Good catch, I think I missed this one, will add it.